synapseutils/sync.py

```diff
@@ -69,6 +69,11 @@ def syncFromSynapse(syn, entity, path=None, ifcollision='overwrite.local', allFi
     :param downloadFile Determines whether downloading the files.
                         Defaults to True
 
+    :param onlyMatching Determines list of regexes to be matched against files.
+                        Only if at least one file matches the regex, it will
+                        be downloaded.
+                        Defaults to None
```

Comment:
What do you think of simplifying this argument to a single regex since multiple patterns can be combined using `|`? My rationale is that we can use [`re.compile()`](https://docs.python.org/3.10/library/re.html#re.compile) downstream on a single expression to be efficient. I'm guessing evaluating a single pre-compiled regex will be faster than compiling and matching a regex (which is what happens when you use `re.match()`) in a for-loop.

Reply:
We could either expect a pre-compiled regex or perform the compilation in `syncFromSynapse()`.

Reply:
Can we think of additional use cases other than filtering on the entity name? I'm just wondering if we can generalize this parameter to being a dictionary mapping entity properties/annotations (keys) to pre-compiled regular expressions (values)? 

@thomasyu888: What do you think?

Reply:
Shifting conversation to here: https://sagebionetworks.jira.com/browse/SYNPY-1236.  @vpchung will be tackling this when she has the time.